soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala

```diff
@@ -573,70 +573,84 @@ private class Merger[T](andFunction: MonomorphicFunction[T]) {
-                        groupBys = aGroup,
-                        having = aHaving,
-                        orderBys = aOrder,
-                        limit = newLim,
-                        offset = newOff,
-                        search = aSearch))
-    case (SoQLAnalysis(false, false, aSelect, Nil, aWhere, Nil, None, aOrder, None, None, aSearch),
-          SoQLAnalysis(false, false, bSelect, bJoins, bWhere, Nil, None, bOrder, bLim, bOff, None)) =>
-      // Can merge a change of filter or order only if no window was specified on the left
-      Some(SoQLAnalysis(isGrouped = false,
-                        distinct = false,
-                        selection = mergeSelection(aSelect, bSelect),
-                        joins = bJoins,
-                        where = mergeWhereLike(aSelect, aWhere, bWhere),
-                        groupBys = Nil,
-                        having = None,
-                        orderBys = mergeOrderBy(aSelect, aOrder, bOrder),
-                        limit = bLim,
-                        offset = bOff,
-                        search = aSearch))
-    case (SoQLAnalysis(false, false, aSelect, Nil, aWhere, Nil,     None,    _,      None, None, aSearch),
-          SoQLAnalysis(true, false, bSelect, bJoins, bWhere, bGroup, bHaving, bOrder, bLim, bOff, None)) =>
-      // an aggregate on a non-aggregate
-      Some(SoQLAnalysis(isGrouped = true,
-                        distinct = false,
-                        selection = mergeSelection(aSelect, bSelect),
-                        joins = bJoins,
-                        where = mergeWhereLike(aSelect, aWhere, bWhere),
-                        groupBys = mergeGroupBy(aSelect, bGroup),
-                        having = mergeWhereLike(aSelect, None, bHaving),
-                        orderBys = mergeOrderBy(aSelect, Nil, bOrder),
-                        limit = bLim,
-                        offset = bOff,
-                        search = aSearch))
-    case (SoQLAnalysis(true,  false, aSelect, Nil, aWhere, aGroup, aHaving, aOrder, None, None, aSearch),
-          SoQLAnalysis(false, false, bSelect, Nil, bWhere, Nil,      None,    bOrder, bLim, bOff, None)) =>
-      // a non-aggregate on an aggregate -- merge the WHERE of the second with the HAVING of the first
-      Some(SoQLAnalysis(isGrouped = true,
-                        distinct = false,
-                        selection = mergeSelection(aSelect, bSelect),
-                        joins = Nil,
-                        where = aWhere,
-                        groupBys = aGroup,
-                        having = mergeWhereLike(aSelect, aHaving, bWhere),
-                        orderBys = mergeOrderBy(aSelect, aOrder, bOrder),
-                        limit = bLim,
-                        offset = bOff,
-                        search = aSearch))
-    case (_, _) =>
-      None
+  private def tryMerge(a: Analysis, b: Analysis): Option[Analysis] = {
+    if (hasWindowFunction(a)) { None }
```

Comment:
Would it make more sense to pull this into the existing match block rather than adding a second branching point? I'm thinking like just adding a basic `case (a, _) if hasWindowFunction(a) => None` to the top